package-builder/flox-build.mk

```diff
@@ -521,43 +553,57 @@ define MANIFEST_BUILD_template =
   # Iterate through this list, replacing all instances of "${package}" with the
   # corresponding storePath as identified by the result-* symlink.
   .PRECIOUS: $($(_pvarname)_buildScript)
-  $($(_pvarname)_buildScript): $(build) check_BUILD_PREREQUISITES
+  $($(_pvarname)_buildScript): $(build) $(PROJECT_TMPDIR)/check-build-prerequisites
+	$(_V_) $(_mkdir) -p $$(@D)
 	@echo "Rendering $(_pname) build script to $$@"
 	@# Always echo lines in the build script as they are invoked.
 	$(_V_) echo "set -x" > $$@.new
 	$(_V_) $(_cat) $$< >> $$@.new
 	$$(eval _sed_args =)
-	$$(foreach _result_link,$$(filter-out $(build) check_BUILD_PREREQUISITES,$$^), \
-	  $$(eval _pkgname = $$(subst -,.,$$(patsubst result-%,%,$$(_result_link)))) \
-	  $$(eval _outpath = $$(shell $(_readlink) $$(patsubst %-out,%,$$(_result_link)))) \
-	  $$(if $$(_outpath),,$$(error could not identify storepath for package $$(dollar){$$(_pkgname)})) \
-	  $$(eval _sed_args += -e 's%$$$$(dollar){$$(_pkgname)}%$$(_outpath)%g'))
-	$$(if $$(_sed_args),$(_V_) $(_sed) -i $$(_sed_args) $$@.new)
+	$$(foreach _buildMetaJSON,$$(filter-out $(build) $(PROJECT_TMPDIR)/check-build-prerequisites,$$^), \
+	  $$(eval _sed_args += $$(shell $(_jq) -r ' \
+	    .pname as $$$$pname | \
+	    .outputs | to_entries | map ( \
+	      " -e " + ("s%\\$$$$$$$${\($$$$pname).\(.key)}%\(.value)%g" | @sh) + \
+	      ( \
+	        if .key == "out" then ( \
+	          " -e " + ("s%\\$$$$$$$${\($$$$pname)}%\(.value)%g" | @sh) \
+		) else "" end \
+	      ) \
+	    )[]' $$(_buildMetaJSON))))
+	$$(if $$(_sed_args),$(_V_) $(_sed) -i $$(strip $$(_sed_args)) $$@.new)
 	$(_V_) $(_mv) -f $$@.new $$@
```

Comment:
This is quite a mess to be honest, nesting three templating languages is not intuitive to read (or edit)